### ISSUE TITLE
Could you strip also titlebars from terminal?

### ISSUE THREAD

**Issue by lpryszcz** [velitasali/GTKTitleBar#14]
First of all, great extension, thanks! Would it be possible to strip title bars from terminal as well? 

![image](https://user-images.githubusercontent.com/988152/59492840-188c7500-8e8a-11e9-8701-e68f50862249.png)

Cheers,

**Comment by velitasali** [comment]
Hey,

Since gnome-terminal is a GTK app, I think it can only be achieved by adding a whitelist/blacklist functionality to the extension.

**Comment by lightbulbjim** [comment]
Intererestingly, the Unite extension (which GTKTitleBar seems to borrow its functionality from) hides the terminal titlebars correctly. Is there some difference in the way they look for windows to undecorate? I did a quick comparison of `modules/windowDecoration.js` in both extensions but nothing jumped out at me.

**Comment by velitasali** [comment]
> Intererestingly, the Unite extension (which GTKTitleBar seems to borrow its functionality from) hides the terminal titlebars correctly. Is there some difference in the way they look for windows to undecorate? I did a quick comparison of `modules/windowDecoration.js` in both extensions but nothing jumped out at me.

Yes, that's true. Actually, this extension is a fork of the Unite extension, but I don't know how it handles the terminal title bar. I should probably look it up if you are saying it can do it without whitelisting etc. 

It can also be programmed into the code just to handle GNOME Terminal

**Comment by paulodiovani** [comment]
> Intererestingly, the Unite extension (which GTKTitleBar seems to borrow its functionality from) hides the terminal titlebars correctly

Does it I tested in Gnome 3.32 and it doesn't seem to work.

Either way, I would also like to have terminal and other Gnome apps titlebars hidden.

**Comment by paulodiovani** [comment]
You can disable gnome terminal _headerbar_ using `gsettings`. This extensions works once it is disabled. :tada: 
Remember to close all terminals.

`gsettings set org.gnome.Terminal.Legacy.Settings headerbar false`

from: https://github.com/franglais125/no-title-bar/issues/114#issuecomment-506085763

**Comment by lightbulbjim** [comment]
> You can disable gnome terminal _headerbar_ using `gsettings`. This extensions works once it is disabled. 🎉
> Remember to close all terminals.
> 
> `gsettings set org.gnome.Terminal.Legacy.Settings headerbar false`
> 
> from: [franglais125/no-title-bar#114 (comment)](https://github.com/franglais125/no-title-bar/issues/114#issuecomment-506085763)

Doesn't work for me on Gnome 3.32.2.

To clarify what I'm seeing:
- I have `org.gnome.Terminal.Legacy.Settings headerbar` set to false.
- GTKTitleBar hides the titlebar on all apps except for gnome-terminal.
- Unite hides the titlebar on all apps, including gnome-terminal.

I disabled all other extensions while testing this, and rebooted in between tests to get things into a known state.

EDIT: Forgot to mention I'm running Wayland. Not sure if that makes a difference.

**Comment by paulodiovani** [comment]
@lightbulbjim It works for 3.32.2, I'm using it.

![image](https://user-images.githubusercontent.com/1295197/60735076-57ad6380-9f28-11e9-9d89-e891a10831a1.png)

So I suppose your problem is due to Wayland.

- Make sure your gnome terminal is NOT using headbar. It should look like this when not maximized:
  ![image](https://user-images.githubusercontent.com/1295197/60734995-1e74f380-9f28-11e9-8c6a-a2996382d62b.png)
- Does it work with other apps even under Wayland?
- Please, test using X11.

**Comment by lightbulbjim** [comment]
Yes, I have gnome-terminal configured to use titlebars. Please see me previous comment.

Good call on Wayland being the problem. Not sure why I didn't think of doing so previously, but I just tested on X11 and the gnome-terminal titlebar is hidden on maximise.

All other apps that I've tested work properly on Wayland (titlebars hide when maximised) except for OpenTTD, which I only noticed today.

So to summarise:
  - Under X11, all apps that I've tested hide their titlebars when maximised.
  - Under Wayland, all apps that I've tested hide their titlebars when maximised except for gnome-terminal and OpenTTD.
  - Using Unite on Wayland _does_ hide the gnome-terminal titlebar when maximised.
  - Unite does _not_ hide the OpenTTD titlebar on Wayland, suggesting that it may be treating gnome-terminal as a special case.

**Comment by velitasali** [comment]
> * Using Unite on Wayland _does_ hide the gnome-terminal titlebar when maximised.
> * Unite does _not_ hide the OpenTTD titlebar on Wayland, suggesting that it may be treating gnome-terminal as a special case.

It seems to be the case. Probably I should check the source code of Unite to figure out how. I haven't really worked on this extension since I first released it.  

If it is the case that Unite treats GNOME Terminal differently then I will try to implement whitelist or similar feature as stated in this issue https://github.com/velitasali/GTKTitleBar/pull/12

**Comment by paulodiovani** [comment]
@velitasali Both unite and pixel-saver have had performance issues in some versions (cpu at 100%).
This extension, however, seems to be lighter due to simplicity.

Please, mind to keep it simple and lightweight. ;)
